### PR TITLE
Update how channel is created in ApplicationsInfo.

### PR DESCRIPTION
**Pull request by hmlanigan**
Create a full channel name with track and risk for CharmHub charms.
Fallback to the old channel which is really a risk.

## QA steps

```console
$ juju add-model seven --config charmhub-url="https://api.staging.charmhub.io"
$ juju deploy ubuntu
Located charm "ubuntu" in charm-hub, revision 18
Deploying "ubuntu" from charm-hub charm "ubuntu", revision 18 in channel stable
$ juju deploy ubuntu-juju-qa --channel 2.0/edge
Located charm "ubuntu-juju-qa" in charm-hub, revision 13
Deploying "ubuntu-juju-qa" from charm-hub charm "ubuntu-juju-qa", revision 13 in channel 2.0/edge
$ juju show-application ubuntu-juju-qa
ubuntu-juju-qa:
  charm: ubuntu-juju-qa
  series: bionic
  channel: 2.0/edge
  constraints:
    arch: amd64
  principal: true
  exposed: false
  remote: false
  endpoint-bindings:
    "": alpha
$ juju show-application ubuntu
ubuntu:
  charm: ubuntu
  series: xenial
  channel: stable
  constraints:
    arch: amd64
  principal: true
  exposed: false
  remote: false
  endpoint-bindings:
    "": alpha
```

### CODE REVIEW

File: apiserver/facades/client/application/application.go [hunk 1]
```diff
@@ -2751,11 +2751,22 @@ func (api *APIBase) ApplicationsInfo(in params.Entities) (params.ApplicationInfo
 			continue
 		}
 
+		var channel string
+		origin := app.CharmOrigin()
+		if origin != nil && origin.Channel != nil {
+			if origin.Channel.Track != "" {
+				channel = fmt.Sprintf("%s/", origin.Channel.Track)
+			}
+			channel = fmt.Sprintf("%s%s", channel, origin.Channel.Risk)
```
Comment (SimonRichardson) [inline apiserver/facades/client/application/application.go:2760]: Can we use `corecharm.Channel{Track: origin.Channel.Track, Risk: origin.Channel.Risk}`, then we can use `Normalize()` and then `String()`. That way we don't duplicate the code here and there.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]
Ace

File: core/charm/channel.go [hunk 2]
```diff
@@ -94,6 +94,17 @@ func MakeChannel(track, risk, branch string) (Channel, error) {
 	}, nil
 }
 
+// MakePermissiveChannel creates a normalized core charm channel which
+// never fails.  It assumes that the risk has been prechecked.
+func MakePermissiveChannel(track, risk, branch string) Channel {
```
Comment (SimonRichardson) [inline core/charm/channel.go:99]: Test?

**Review by SimonRichardson — APPROVED** [review]
Can we add a test? Otherwise, Q&A went well.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**